Modify rule S2631(Java): Update text to Education Framework structure (APPSEC-126) (#1223)
SonarSource/rspec: rules/S2631/java/how-to-fix-it/java-se.adoc

=== How to fix it in Java SE

The following non-compliant code is vulnerable to Regex Denial of Service

> because untrusted data is used as a pattern to scan a string without prior

Reviewer: ```suggestion
because untrusted data is used as a regex to scan a string without prior
```

sanitization or validation.

[cols="a"]
|===
h| Non-compliant code example
|
[source,java]
----
public boolean validate(HttpServletRequest request) {
  String regex = request.getParameter("regex");
  String input = request.getParameter("input");

  input.matches(regex);  // Noncompliant
}
----
h| Compliant solution
|
[source,java]
----
public boolean validate(javax.servlet.http.HttpServletRequest request) {
  String regex = request.getParameter("regex");
  String input = request.getParameter("input");

>   input.matches(Pattern.quote(regex));

Reviewer: ```suggestion
  return input.matches(Pattern.quote(regex));
```

}
----
|===

=== How does this work?

include::../../common/fix/validation.adoc[]

> In the example, `Patter.quote` escapes metacharacters and escape sequences that

Reviewer: ```suggestion
In the compliant solution example, `Pattern.quote` escapes metacharacters and escape sequences that
```

Author: yeah !

Author: done here: https://github.com/SonarSource/rspec/pull/1218/commits/03ad33abf4c4cf6f2ead4daf56c79d24b0159670
